google/data_source_google_service_account_id_token_test.go

```diff
@@ -0,0 +1,78 @@
+	"client_email": "user@gcp-project.iam.gserviceaccount.com",
+	"client_id": "103198861025845558729",
+	"auth_uri": "https://accounts.google.com/o/oauth2/auth",
+	"token_uri": "https://accounts.google.com/o/oauth2/token",
+	"auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+	"client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/user%40gcp-project.iam.gserviceaccount.com"
+  }
+  `
+
+const targetAudience = "https://foo.bar/"
+const fakeIdToken = "eyJhbGciOiJSUzI1NiIsIm..."
+
+func testAccCheckServiceAccountIdTokenValue(name, value string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		ms := s.RootModule()
+		rs, ok := ms.Outputs[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		// TODO: validate the token belongs to the service account
+		if rs.Value == "" {
+			return fmt.Errorf("%s Cannot be empty", name)
+		}
+
+		return nil
+	}
+}
+
+func TestAccDataSourceGoogleServiceAccountIdToken_basic(t *testing.T) {
+	t.Parallel()
+
+	resourceName := "data.google_service_account_id_token.default"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config:  testAccCheckGoogleServiceAccountIdToken_datasource(targetAudience),
+				Destroy: true,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "target_audience", targetAudience),
+					testAccCheckServiceAccountIdTokenValue("oidc-token", fakeIdToken),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckGoogleServiceAccountIdToken_datasource(targetAudience string) string {
+
+	return fmt.Sprintf(`
+data "google_service_account_id_token" "default" {
+  target_audience = "%s"
+}
+
+output "oidc-token" {
```

Comment:
You shouldn't need to have the output in the test, you can just verify by checking the data source directly

Reply:
ok, i removed the explicit 'output` step, PTAL